monolith/views/test/test_stories.py

```diff
@@ -91,7 +98,83 @@ def test_new_story(self):
         reply = self.client.post('/stories/new_story', data={ "theme" : "Mountain", "dice_number" : "3"})
         self.assertEqual(reply.status_code, 302)
+        with self.context:
+            s = Story.query.filter_by(id=3).first()
+            self.assertIsNone(s)
+            
+    def test_write_story(self):
+        
+        # error: story id not valid
+        self._login("example@example.com", "admin")
+        reply = self.client.get('/write_story/3')
+        self.assertEqual(reply.status_code, 404)
+        
+        # error: I'm not author
+        self._logout()
+        self._signup("fantastic@example.com", "betterNerfIrelia", "404", "404", "01/01/1964", True)
+        self._login("fantastic@example.com", "betterNerfIrelia")
+        reply = self.client.post('/stories/new_story', data={ "theme" : "Mountain", "dice_number" : "3"})
+        with self.context:
+            s = Story.query.filter_by(id=2).first()
+            self.assertIsNotNone(s)
+        self._logout()
+        self._login("example@example.com", "admin")
+        reply = self.client.get('/write_story/2')
+        self.assertEqual(reply.status_code, 401)
+        
+        # success: render write_story.html
+        self._logout()
+        self._login("fantastic@example.com", "betterNerfIrelia")
+        reply = self.client.get('/write_story/2')
+        self.assertEqual(reply.status_code, 200)
+        self.assert_template_used("/write_story.html")
+
+        # error: title needed
+        reply = self.client.post('/write_story/2', data={
+                 'text' : "Mountain",
+                 'title' : "",
+                 'store_story' : 1})
+        self.assert_context("message", "You must complete the title in order to publish the story")
+        self.assertEqual(reply.status_code, 200)
+
+        # error: story not valid
+        reply = self.client.get('/write_story/2')
+        self.assertEqual(reply.status_code, 200)
+        reply = self.client.post('/write_story/2', data={
+                 'text' : "1",
+                 'title' : "ThisIsATitle",
+                 'store_story' : 1})
+        self.assert_context("message", "You must use all the words of the outcome!")
+        
+        # success: written draft
+        reply = self.client.get('/write_story/2')
+        self.assertEqual(reply.status_code, 200)
+        reply = self.client.post('/write_story/2', data={
+                 'text' : "1",
+                 'title' : "ThisIsATitle",
+                 'store_story' : 0})
+        self.assert_template_used("/write_story.html")
+        
+        # TODO
+        # success: published story
+        #reply = self.client.get('/write_story/2')
+        #self.assertEqual(reply.status_code, 200)
+        #reply = self.client.post('/write_story/2', data={
+        #         'text' : "1",
+        #         'title' : "ThisIsATitle",
+        #         'store_story' : 0})
+        #self.assert_template_used("/write_story.html")
```

Comment:
same here

Reply:
Ok I was hoping to get it to work but I'll delete them for now